### PR TITLE
s3gw install doc update

### LINKED ISSUE #192: docs: updates to VM instructions
### What needs to be done

* Update with correct URL to UI: [https://github.com/aquarist-labs/s3gw/blob/main/tools/env/setup-vm.sh](https://github.com/35f7292551d54f42af992b9ad770e6cb) - Currently it's pointing to the experimental UI. Now we're developing our own here: https://github.com/aquarist-labs/s3gw-ui
* Update references to k8s in documentation: https://s3gw-docs.readthedocs.io/en/latest/s3gw-with-k8s-k3s/#building-the-environment (`./setup-k8s.sh`) doesn't exist anymore.

### LINKED ISSUE #251: docs: Documentation revision
## What needs to be done

Do a full revision of our documentation, find out which information is missing, and which is out of date and correct it if needed.

## Why it needs to be done

The Rancher Communities folks, who can help us promote the project through different actions, require the project to have well-defined documentation. We will review our documentation on a regular basis.

### PR DESCRIPTION
**Pull request by asettle**
# Describe your changes

Updating the documentation to reflect latest installation procedure

## Issue ticket number and link

Fixes: #251, #192

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] CHANGELOG.md has been updated should there be relevant changes in this PR.

### CODE REVIEW

File: docs/helm-charts.md [hunk 1]
```diff
@@ -1,82 +1,42 @@
-# Installation and options
+# Installing s3gw with helm charts
```
Comment (votdev) [inline docs/helm-charts.md]:
```suggestion
# Installing s3gw with Helm charts
```

File: docs/helm-charts.md [hunk 2]
```diff
@@ -1,82 +1,42 @@
...
 
-The canonical way to install the helm chart is via a helm repository:
+Before you begin, ensure you install helm. To install, see the [documentaiton](https://helm.sh/docs/intro/install/)
```
Comment (votdev) [inline docs/helm-charts.md]:
```suggestion
Before you begin, ensure you install Helm. To install, see the [documentaiton](https://helm.sh/docs/intro/install/)
```

And following

**Review by votdev — CHANGES_REQUESTED** [review]

**Comment by asettle** [comment]
TOC has been updated to reflect the following output:

Screenshot 2023-06-19 at 14 32 01

Two new files are now included: configuration and troubleshooting

File: docs/config-s3gw.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+      crt: CERTIFICATE_FOR_UI
+      key: CERTIFICATE_KEY_FOR_UI
+```
+
+The certificates must be provided as base64 encoded PEM in one long
+string without line breaks. You can create them from a PEM file.
+
+## s3gw user interface
+
+`ui.enabled` is by default set to `true`, but can be set to false to use
```
Comment (votdev) [inline docs/config-s3gw.md]:
```suggestion
`ui.enabled` is by default set to `true`, but can be set to `false` to use
```

File: docs/config-s3gw.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+### Local storage
+
+You can use the `storageClass.local` and `storageClass.localPath` variables to
+set up a node-local volume for testing if you don not have Longhorn. This is an
+experimental feature for development use only.
+
+## Log settings
+
+The log verbosity can also be configured for the s3gw pod. Set the `logLevel`
+property to a number, with `"0"` being the least verbose and higher numbers
```
Comment (votdev) [inline docs/config-s3gw.md:210]: What is the highest possible `logLevel` value?

File: docs/s3gw-with-k8s.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+<!-- markdownlint-disable no-duplicate-heading -->
+# Installing k8s for s3gw
+
+The following document describes the prerequisite installations to run the
+S3 Gateway (s3gw), an S3 object storage service, on top of a vanilla k8s
```
Comment (votdev) [inline docs/s3gw-with-k8s.md]:
```suggestion
S3 Gateway (s3gw), an S3 object storage service, on top of a vanilla K8s
```

And following ...

**Review by votdev — DISMISSED** [review]

File: docs/s3gw-with-k8s.md (hunk 5 above)
Comment (asettle) [reply docs/s3gw-with-k8s.md]: I need to fix this throughout, cheers

File: docs/config-s3gw.md (hunk 4 above)
Comment (asettle) [reply docs/config-s3gw.md:210]: @m-ildefons do we have one or is this currently infinite? Should we set a recommended level?

File: docs/troubleshoot-s3gw.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+
+# Troubleshooting the s3w
```
Comment (m-ildefons) [inline docs/troubleshoot-s3gw.md]: Typo

File: docs/s3gw-with-k8s.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+### Access the Longhorn UI
+
+Now that you have installed Longhorn, access the localhost UI:
+`http://longhorn.<LOCAL-ADDRESS>`.
+
+You should now be able to see Longhorn running and there should be no volumes
+present.
+
+## Install certification manager
```
Comment (m-ildefons) [inline docs/s3gw-with-k8s.md]: Either `certificate manager` or `cert-manager`, but `certification` has nothing to do with this.

File: docs/s3gw-with-k3s.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+### Access the Longhorn UI
+
+Now that you have installed Longhorn, access the localhost UI:
+`http://longhorn.<LOCAL-ADDRESS>`.
+
+You should now be able to see Longhorn running and there should be no volumes
+present.
+
+## Install certification manager
```
Comment (m-ildefons) [inline docs/s3gw-with-k3s.md]: Either `certificate manager` or `cert-manager`, but `certification` has nothing to do with this.

File: docs/config-s3gw.md (hunk 4 above)
Comment (m-ildefons) [reply docs/config-s3gw.md:210]: This is the same as Ceph log levels. Their docs suggest they go at least up to 30, but I think any integer value is valid. We might just say 10 is the highest as we probably won't utilize any higher log level in any meaningful way.

File: docs/config-s3gw.md [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+cluster and from the outside. For both, the s3gw must be configured with the
+correct service and domain name. Use the `publicDomain` and the
+`ui.publicDomain` setting to configure the domain under which the s3gw and the
+UI respectively be available to the outside of the Kubernetes cluster. Use the
+`privateDomain` setting to set the cluster's internal domain and make the s3gw
+available inside the cluster to other deployments.
+
+```yaml
+serviceName: s3gw
+publicDomain: "fe.127.0.0.1.omg.howdoi.website"
```
Comment (m-ildefons) [inline docs/config-s3gw.md]: I'd prefer to see an `example.com` domain here.

File: docs/config-s3gw.md [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+privateDomain: "s3gw-namespace.svc.cluster.local"
+ui:
+  serviceName: s3gw-ui
+  publicDomain: "be.127.0.0.1.omg.howdoi.website"
```
Comment (m-ildefons) [inline docs/config-s3gw.md]: See above

File: docs/s3gw-with-k8s.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
+<!-- markdownlint-disable no-duplicate-heading -->
+# Installing K8s for s3gw
+
+The following document describes the prerequisite installations to run the
+S3 Gateway (s3gw), an S3 object storage service, on top of a vanilla K8s
```
Comment (m-ildefons) [inline docs/s3gw-with-k8s.md:5]: "vanilla K8s" is the colloquial name I use for the K8s you get when you install them from Google's repos using `kubeadm`:
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/
Note that it's probably quite different than what you'd get e.g. at a hosted K8s provider or when using other installers like the `kubespray` Ansible playbooks.

**Review by m-ildefons — DISMISSED** [review]
Some small nits, but LGTM.

File: docs/s3gw-with-k8s.md (hunk 11 above)
Comment (asettle) [reply docs/s3gw-with-k8s.md:5]: I actually liked the term and wanted to float it. It tends to lean more towards a vague "you can just use k8s here" - and the concept of "vanilla" is understood.

Comment (asettle) [reply docs/s3gw-with-k8s.md:5]: To be clear, open to other ways of phrasing it: I just reviewed other terms and I liked this.

Comment (m-ildefons) [reply docs/s3gw-with-k8s.md:5]: It'll be fine. I'm just saying that it's quite hand-wavy and not very well defined.

File: docs/config-s3gw.md [hunk 12]
```diff
@@ -0,0 +1,322 @@
...
+storageClass:
+  name: my-custom-storageclass
+  create: false
+```
+
+### Local storage
+
+You can use the `storageClass.local` and `storageClass.localPath` variables to
+set up a node-local volume for testing if you don not have Longhorn. This is an
+experimental feature for development use only.
```
Comment (callithea) [inline docs/config-s3gw.md]: Maybe the "experimental feature"-sentence could be put as a ``**NOTE::**``?

Comment (m-ildefons) [reply docs/config-s3gw.md]: Also there's a typo here: `... don not have Longhorn...`

**Review by jhmarina — COMMENTED** [review]
lgtm - other than the 2 things mentioned above 😃

File: docs/config-s3gw.md (hunk 12 above)
Comment (asettle) [reply docs/config-s3gw.md]: don is good

**Review by callithea — DISMISSED** [review]
Lgtm

File: docs/s3gw-with-k3s.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+You can use a different ingress controller, but note you will have to
+create your own ingress resource.
+
+## Installing k3s
+
+Before you begin, ensure you install K3s. You can find the installation
+instructions [here](https://k3s.io/) or, run:
+
+```bash
+curl -sfL https://get.k3s.io | sh -.
```
Comment (callithea) [inline docs/s3gw-with-k3s.md]:
```suggestion
curl -sfL https://get.k3s.io | sh -
```

**Review by Ameenasuhani — COMMENTED** [review]
https://github.com/aquarist-labs/s3gw/blob/7cb05a7efe81423193e431e7d6a7bdfdd07ac804/docs/ui.md?plain=1#L19
then hyperlink is not working in ui doc
`s3gw-with-k8s-k3s.md/#k3s-with-longhorn` fix it to this

**Review by jhmarina — APPROVED** [review]
lgtm

**Review by callithea — APPROVED** [review]
Lgtm 🙂

**Pull request merged**